Sources/Internal/GrpcClient.swift

```diff
@@ -274,44 +218,39 @@ actor GrpcClient: CustomStringConvertible {
     do {
       if let token = try await auth.currentUser?.getIDToken() {
         headers.add(name: RequestHeaders.authorizationHeader, value: "\(token)")
-        if debugEnable == true {
-          FirebaseLogger.dataConnect
-            .debug("Auth token added: \(token)")
-        } else {
-          FirebaseLogger.dataConnect
-            .debug("Auth token added.")
-        }
+        DataConnectLogger.debug("Auth token added.")
       } else {
-        FirebaseLogger.dataConnect.debug("No auth token available. Not adding auth header.")
+        DataConnectLogger.debug("No auth token available. Not adding auth header.")
       }
     } catch {
-      FirebaseLogger.dataConnect
+      DataConnectLogger
         .debug("Cannot get auth token successfully due to: \(error). Not adding auth header.")
     }
 
     // Add AppCheck token if available
     do {
       if let token = try await appCheck?.token(forcingRefresh: false) {
         headers.add(name: RequestHeaders.appCheckHeader, value: token.token)
-        if debugEnable == true {
-          FirebaseLogger.dataConnect
-            .debug("App Check token added: \(token.token)")
-        } else {
-          FirebaseLogger.dataConnect
-            .debug("App Check token added.")
-        }
+        DataConnectLogger.debug("App Check token added.")
       } else {
-        FirebaseLogger.dataConnect
-          .debug("App Check token unavailable. Not adding App Check header.")
+        DataConnectLogger.debug("App Check token unavailable. Not adding App Check header.")
       }
     } catch {
-      FirebaseLogger.dataConnect
-        .debug(
-          "Cannot get App Check token successfully due to: \(error). Not adding App Check header."
-        )
+      DataConnectLogger.debug(
+        "Cannot get App Check token successfully due to: \(error). Not adding App Check header."
+      )
+    }
+
+    var options = CallOptions(customMetadata: headers)
+
+    // Enable GRPC tracing
+    if DataConnectLogger.logLevel.rawValue >= FirebaseLoggerLevel.debug.rawValue,
+       DataConnect.ArgumentFlag.privateLoggingEnabled == false {
+      var logger = Logger(label: "com.google.firebase.dataconnect.grpc")
+      logger.logLevel = .trace
+      options.logger = logger
```

Comment:
Why not pass a `DataConnectLogger` instance instead?

Reply:
The logger option in grpc tracing is from `Logging` package, `DataConnectLogger` is from osLog